src/core/readyException.js

```diff
@@ -0,0 +1,13 @@
+define( [
+	"../core"
+], function( jQuery ) {
+
+"use strict";
+
+jQuery.readyException = function( error ) {
+	window.setTimeout( function() {
+		throw error;
+	} );
```

Comment:
The important detail to me is that our default handling doesn't suppress unhandled rejections, since doing so was the essence of the original complaint (and I still want to keep the door open for an unhandled rejections plugin). In fact, if we were willing to advocate `$.when($.ready).then(fn).catch(errfn)` strongly enough, then gh-3174 could be closed with no code changes whatsoever.